src/sniffer/pkg/collectors/socketscanner.go

```diff
@@ -39,7 +39,9 @@ func (s *SocketScanner) scanTcpFile(hostname string, path string) {
 		}
 
 		if _, ok := listenPorts[sock.LocalAddr.Port]; ok {
-			s.addCapturedRequest(sock.RemoteAddr.IP.String(), hostname, sock.LocalAddr.IP.String(), time.Now())
+			// FIXME: don't check hostname since we have the server's hostname and the client's IP here
+			// For example, Remote: 10.244.120.96 (loadgenerator) -> Local: 10.244.120.95 (frontend). The hostname we have is frontend (local), but the client we are attempting to report is loadgenerator.
+			s.addCapturedRequest(sock.RemoteAddr.IP.String(), "", sock.LocalAddr.IP.String(), time.Now())
```

Comment:
Yes the hostname would always be for the local address...
It confused me because I just thought that for some reason "RemoteAddr = SrcAddr" & "LocalAddr = DstAddr".

The thing I don't get - there's no information about which is the client and which is the server... so the traffic could be reported at the opposite direction. Or does `/proc/net/tcp` only shows incoming connections?

Reply:
Seems like this information (whether incoming/outgoing) is not included in `/proc/net/tcp`.
The only way I see to do it is to check if the local port shows up in the "Listening" entries.
In every conceivable scenario - there's still gonna be a listening socket while an incoming TCP connection exist).